releasenotes/notes/fix-operator-matrix-multiplication-947c2813f14ce402.yaml

```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    The :meth:`.Operator.__matmul__` method now correctly handles
+    :class:`.Statevector` inputs by delegating to
+    :meth:`.Statevector.evolve`, ensuring consistent behavior with
+    operator–statevector application, and returning a :class:`.Statevector` of the
+    appropriate dimension and mathematical matrix multiplication value. Fixes Issue #14726.
```

Comment:
just a minor comment - can you please make 14726 into a link? should look like:
See `#14726 <https://github.com/Qiskit/qiskit/pull/14726>`__ for details.